shared/stateutil/state_root_test.go

```diff
@@ -0,0 +1,55 @@
+package stateutil_test
+
+import (
+	"testing"
+
+	"github.com/prysmaticlabs/go-ssz"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/interop"
+	"github.com/prysmaticlabs/prysm/shared/stateutil"
+)
+
+func TestHashTreeRootEquality(t *testing.T) {
```

Comment:
Would it be possible to test everything else? Benchmarks aren't needed but maybe splitting it up would make the tests cleaner too.